### PR TITLE
{WIP} Update time range for IKStudy analysis

### PR DESCRIPTION
**Pull request by jimmyDunne**
### Fixes issue 
User input time is not respected by the IK Study. This is most likely caused due to the times being defined directly from the orientationsTable and (seemingly) not from the user-specified times. 

```
...
auto& times = oRefs.getTimes();

s0.updTime() = times[0];
    ikSolver.assemble(s0);
    if (visualizeResults) {
        model.getVisualizer().show(s0);
        model.getVisualizer().getSimbodyVisualizer().setShowSimTime(true);
    }
    for (auto time : times) {
        s0.updTime() = time;
...
```

### Brief summary of changes
I added a line that trim's the orientationTable based on user specified times. This seemed to be the quickest/easiest way for me to get the user-specified times working locally. 

### Testing I've completed
Run locally (through Matlab) on my own files. 

### Looking for feedback on...
I doubt this is the most robust way to solve this bug, but was the easiest for me to code. Happy to improve :)
### CHANGELOG.md (choose one)

- no need to update because...
- updated...

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2621)

### CODE REVIEW

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp [hunk 1]
```diff
@@ -143,7 +143,10 @@ runInverseKinematicsWithOrientationsFromFile(Model& model,
 
     TimeSeriesTable_<SimTK::Rotation> orientationsData =
         OpenSenseUtilities::convertQuaternionsToRotations(quatTable);
-
+    
+    // Trim orientation data based on User input times 
+    orientationsData.trim(get_time_range(0),get_time_range(1));
```
Comment (tkuchida) [inline OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: May want to check that `orientationsData` is non-empty after trim. It doesn't look like [`trim()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/TimeSeriesTable.h#L440) guarantees this as a post condition. Also, `trim()` calls [`trimToIndices()`](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/TimeSeriesTable.h#L492), which notes that "Side effect may include that headers/metaData may be left stale." Not sure whether any of the code below assumes the metadata is correct.

Comment (aymanhab) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: Thanks @tkuchida I agree that having an empty table is likely due to error so I'd leave this change here as is and modify the trim method to throw on empty table so the check is not repeated elsewhere, I'll make the change to this branch to be part of the PR. Typically meta data does not contain info that depends on number of rows and this is the case here.

Comment (aymanhab) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: Ready for review @tkuchida Build failures are due to refactoring on the destination branch so unrelated to this PR

**Comment by jimmyDunne** [comment]
I am still working on this. There are errors if you don't set a time. Currently working on something so will update in ~hour

**Comment by jimmyDunne** [comment]
I had issues when no user specifed time was set. This is because the default time would be `Infinity`. 

Instead, I updated the function `getTimeRangeInUse` to validate and set the time ranges that would be used in the `table.trim()` method. I (think) I have tested for all conditions (Matlab). If the user doesn't set any time then the start and end times are used. 

This change will break the function `previewExperimentalData` at L99, since the `for` loop is expecting indices and not start and end time values. I figure this is okay since we won't (shouldn't) be using the function anyway (it only deals with markers whose use will be removed from this class). 

@aymanhab, can you give me feedback on anything I can change?

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp (hunk 1 above)
Comment (tkuchida) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: Perhaps add test cases to exercise the failure modes and ensure these methods remain well-behaved in the future? I don't know where the tests are for OpenSense's InverseKinematicsStudy, though, as there's no Test directory under OpenSim/Simulation/OpenSense 🤔

Comment (aymanhab) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: These are lumped into the testOpensense test case, as we refactor we should break this into more modular tests.

**Comment by aymanhab** [comment]
@jimmyDunne This feels like replicating the trim functionality somewhere else. The trim functionality is implemented at the TimeSeriesTable level and should be available for Marker data as well. If that doesn't work now please advise and we'll refine it. I'd prefer that over repeating the functionality inside a 'preview...' method

**Comment by aymanhab** [comment]
@jimmyDunne not sure why you closed this instead of using trim but the branch has the improvement to trim that @tkuchida suggested so I'd like to include it at some point. Thank you

**Comment by jimmyDunne** [comment]
I couldn't get Xcode or Clion working so I am trying to write the CPP code in a basic text editor and so any type of debugging or type fixes are taking an exceedingly long time. I have to move on to some other work and don't have the time right now to address the required feedback.

**Comment by aymanhab** [comment]
No worries @jimmyDunne I'll take care of the changes. Thank you

**Pull request closed without merging**